XNLI 2.0 subset added

dokato opened the pull request:
<!-- If you are not submitting for a dataset, feel free to remove the content below  -->
I added extension of XNLI dataset, XNLI 2.0, described in this paper: https://arxiv.org/pdf/2301.06527 for Indian languages.

I collated them into one multilingual set added to: [`mteb/xnli2.0-multi-pair`](https://huggingface.co/datasets/mteb/xnli2.0-multi-pair)
<!-- add additional description, question etc. related to the new dataset -->

## Checklist for adding MMTEB dataset

<!-- 
Before you commit here is a checklist you should complete before submitting
if you are not 
 -->
Reason for dataset addition:
<!-- Add reason for adding dataset here. E.g. it covers task/language/domain previously not covered -->


- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [ ] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

isaac-chung reviewed (approved):
Thanks for adding this! Looks good, can merge after adding points + linting.

dokato commented:
Thanks @isaac-chung! Fixed and added points according to:
```
> tasks = get_tasks(task_types=['PairClassification'])
> len(set(XNLI2().metadata.languages).intersection(tasks.languages))
0
```
which means that none of the languages are yet in this task, so `2+4*10`.
